dynamic_programming/longest_common_string.cpp

```diff
@@ -1,53 +1,142 @@
-#include <iosrteam>
-using namespace std;
+/**
+ * @file
+ * @brief contains the definition of the function longest_common_string_length
```

Comment:
could you add links/references to the sources?
like [this page](https://en.wikipedia.org/wiki/Longest_common_substring)

Reply:
I could not find anything resorbable. Please note, that the wiki page linked by you describes another problem. There the question is about the longest common **substring** (i.e. it has to be present in both of the inputs as a sequence of consecutive characters). Here the optimal string might be interrupted (cf. this [TestCase](https://github.com/TheAlgorithms/C-Plus-Plus/blob/da2567469aa748c877cf314cfcb75598313ca4ca/dynamic_programming/longest_common_string.cpp#L70)).

Reply:
I see thank you for the clarification

Reply:
Pleasure!